core/store/src/trie/trie_storage.rs

```diff
@@ -34,7 +34,10 @@ impl TrieCache {
         self.0.lock().expect(POISONED_LOCK_ERR).clear()
     }
 
-    pub fn update_cache(&self, ops: Vec<(CryptoHash, Option<&Vec<u8>>)>) {
+    pub fn update_cache(&self, ops: Vec<(CryptoHash, Option<&Vec<u8>>)>, shard_uid: ShardUId) {
```

Comment:
Now I think that `TrieCache` should hold `shard_uid`, so we could avoid passing it here. Let's fix it in a separate PR